src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/SetAzureSqlDatabaseBackupLongTermRetentionPolicy.cs

```diff
@@ -41,20 +61,86 @@ public class SetAzureSqlDatabaseBackupLongTermRetentionPolicy : AzureSqlDatabase
         /// Gets or sets the name of the backup long term retention policy
         /// </summary>
         [Parameter(Mandatory = true,
+            ParameterSetName = LegacySet,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "The Resource ID of the backup long term retention policy.")]
         [ValidateNotNullOrEmpty]
         [Alias("Id")]
         public string ResourceId { get; set; }
 
+        /// <summary>
+        /// Gets or sets the Weekly Retention.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            ParameterSetName = WeeklyRetentionRequiredSet,
+            Position = 3,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Weekly Retention of the long term retention policy.")]
+        [Parameter(Mandatory = false,
+            ParameterSetName = MonthlyRetentionRequiredSet,
+            Position = 4,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Weekly Retention of the long term retention policy.")]
+        [Parameter(Mandatory = false,
+            ParameterSetName = YearlyRetentionRequiredSet,
+            Position = 5,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Weekly Retention of the long term retention policy.")]
```

Comment:
Sorry but this explanation is really not clear. From reading this help I have no idea what weekly retention means or what the input should look like. Please write your help text assuming that the reader have absolutely no background knowledge of how the feature works.

Using my insider knowledge, I am aware that these should be iso8601 durations. It would be really nice if there were simple parameter options like `-WeeklyRetentionMonths` or whatever to decrease the barrier to entry.

Reply:
Changed it so the retention parameters also take integers that will be assumed as days.